lib/xandra/cluster.ex

```diff
@@ -425,35 +437,45 @@ defmodule Xandra.Cluster do
   @impl true
   def handle_cast(message, state)
 
-  def handle_cast({:activate, node_ref, address, port}, %__MODULE__{} = state) do
+  def handle_cast({:activate, node_ref, address, port, data_center}, %__MODULE__{} = state) do
     _ = Logger.debug("Control connection for #{:inet.ntoa(address)}:#{port} is up")
 
     # Update the node_refs with the actual address of the control connection node.
-    state = update_in(state.node_refs, &List.keystore(&1, node_ref, 0, {node_ref, address}))
+    state =
+      update_in(
+        state.node_refs,
+        &List.keystore(&1, node_ref, 0, {node_ref, {address, data_center}})
```

Comment:
Lists of tuples can have flat tuples, so I would recommend to store this flat:

```suggestion
        &List.keystore(&1, node_ref, 0, {node_ref, address, data_center})
```